tmk_core/protocol/lufa/adafruit_ble.cpp

```diff
@@ -29,6 +29,11 @@
 #define SAMPLE_BATTERY
 #define ConnectionUpdateInterval 1000 /* milliseconds */
 
+#ifdef SAMPLE_BATTERY
+#include "analog.c"
```

Comment:
I would prefer seeing thes line above.  

Also, shouldn't this be a header file?

Reply:
Same here: I'm trying to touch existing code as little as possible, and since the `#define` was here.... If you'd like, I can move this chunk to the header.

Reply:
This line should be:

```suggestion
#include "analog.h"
```

And yeah, it should go at the top with the rest of the includes, even though it's not used unless `SAMPLE_BATTERY` is defined... which it always is.

Reply:
Now I'm revealing my ignorance of C++. If I use `#include "analog.h"`, I get this error at linking:

    .../adafruit_ble.cpp:651: undefined reference to `analogRead(unsigned char)'

As I have it, everything works. Am I doing something wrong?

In any case, I'm happy to move the `#include` line up; should it be wrapped with `#ifdef SAMPLE_BATTERY ... #endif`? (Should that `#define` be moved to `adafruit_ble.h`?

Reply:
Ah, since `analog.c` is, well a C file, and this is C++, we need to do this:

```cpp
extern "C" {
#include "analog.h"
}
```

Reply:
OK. I've done that, and I'm getting no errors on `make`. Thanks for the help.